### PR TITLE
Add needed ingredient method

### PR DESCRIPTION
**Pull request by cagallo**
#### What's this PR do?
- Finish pantry class (final method-getMissingingredients)
- Fix test data

#### Where should the reviewer start?

- Pantry.js

#### How should this be manually tested?

- npm test

#### Any background context you want to provide?

N/A

### CODE REVIEW

**Review by kammon10 — APPROVED** [review]
awesome work!

File: src/classes/Pantry.js [hunk 1]
```diff
@@ -7,17 +7,38 @@ class Pantry {
     let hasAllIngredients = true;
 
     recipe.ingredients.forEach(recipeIng => {
-      const ingInPantry = this.ingredients.find(pantryIng => {
+      const matchedId = this.ingredients.find(pantryIng => {
```
Comment (aliroemhildt) [inline src/classes/Pantry.js:10]: nice! this naming makes a lot more sense

**Review by aliroemhildt — APPROVED** [review]
looks great!

File: test/Pantry-test.js [hunk 2]
```diff
@@ -34,19 +34,18 @@ describe('Pantry', () => {
...
     const recipe1 = sampleData[0];
     const recipe2 = sampleData[1];
-    expect(user.pantry.getMissingIngredients(recipe1).to.deep.equal([]))
-    expect(user.pantry.getMissingIngredients(recipe2).to.deep.equal([
+    expect(user.pantry.getMissingIngredients(recipe1)).to.deep.equal([]);
+    expect(user.pantry.getMissingIngredients(recipe2)).to.deep.equal([
       {'ingredient': 1009016, 'amount': 1.5},
       {'ingredient': 20027, 'amount': 1},
       {'ingredient': 1002046, 'amount': 1},
-      {'ingredient': 20027, 'amount': 1},
```
Comment (anewb87) [inline test/Pantry-test.js:42]: This was causing a headache. Finally realized we accidentally put it in there twice. Deleted one and viola! It worked.

File: src/classes/Pantry.js [hunk 3]
```diff
@@ -7,17 +7,38 @@ class Pantry {
...
-      if ((ingInPantry && (ingInPantry.amount < recipeIng.quantity.amount)) || !ingInPantry) {
+      if ((matchedId && (matchedId.amount < recipeIng.quantity.amount)) || !matchedId) {
         hasAllIngredients = false;
       }
     })
 
     return hasAllIngredients;
   }
+
+  getMissingIngredients(recipe) {
```
Comment (anewb87) [inline src/classes/Pantry.js:22]: Dig it!

**Review by anewb87 — APPROVED** [review]
Done with Pantry class (in theory)

**Pull request merged**